src/core/datastructures/bitset.cpp

```diff
@@ -143,6 +145,12 @@ bool BitSet::set(const BitSet& b) {
 
 void BitSet::add(util::span<const uint32_t> span) { addMany(span.size(), span.data()); }
 
+void BitSet::add(const std::vector<bool>& v) {
+    for (std::uint32_t i = 0; i < static_cast<std::uint32_t>(v.size()); ++i) {
```

Comment:
Might throw here if v > uint32::max

Did you try the add many version? think that might be quite good here...

Reply:
Wouldn't addMany require a temporary span/vector of indices?